src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2479,7 +2493,8 @@ private void emitProperty(
             && unionType
                 .getAlternatesWithoutStructuralTyping()
                 .stream()
-                .anyMatch(JSType::isVoidType)) {
+                .anyMatch(JSType::isVoidType)
+            && !isNamespace) {
```

Comment:
Took me second to realize that 'var optional?: number' in not a valid ts syntax. Add a comment why isNamespace check is needed for syntactic correctness.

Reply:
Done.